src/lib/utils/size_estimation_utils.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <random>
+
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+  * This function iterates over the given string vector @param string_vector strings and sums up the memory usage. Due
+  * to the small string optimization (SSO) in most current C++ libraries, each string has an initially allocated buffer
+  * (e.g., 15 chars in GCC's libstdc++). If a string is larger, the string is allocated on the heap and the initial
+  * string object stores a pointer to the actual string on the heap.
+  *
+  * Depending on the @param mode, either all strings all considered or a sample is taken and evaluated.
+  *
+  * Please note, that there are still differences between the stdlib's. Thus, even the full size accumulation is not
+  * guaranteed to be 100% accurate for all libraries.
```

Comment:
Not sure how these two things belong together. What has the difference in SSO to do with the accuracy of this?

Reply:
Clang and GCC (or better: their libs) use different ways to implement SSO. The current calculations should be somewhat accurate but I cannot guarantee it (https://shaharmike.com/cpp/std-string/), especially not for upcoming versions.

Reply:
For me, those are two different things. First, the standard does not prescribe an exact allocation strategy, so our approach may silently become invalid in the future. Second, the different implementations do different things and users should be aware of that.

Reply:
So what's your recommendation?

Reply:
`s/Thus, even/Also/` would be enough for me. I'm just bugged by the text suggesting a logical connection where I don't see one.

Reply:
Ah ok. Now I get you. Changed.